### DISCUSSION TITLE
To which file are photon position and direction on exit saved?

### DISCUSSION THREAD

**Discussion by RecurvedBow** [fangq/mcx#236] [General]
Hello everyone,

I would like to save the photon position and direction upon exit. For this, I am using the command flag --saveexit. However, no file is created containing the photon data. Where can I find the data?

For context, my simulation is run with the following args: -f MyVolumeName.json -O F -b 1 -a 1 -F jnii --saveexit -H 10000000 --bc mm_mm_000000 --saveref

My input files are: 
- MyVolumeName.json which defines the illumination (planar), the volume media format (asgn_float) and the volume file path
- MyVolumeName.bin which defines the optical parameters (mu_a, mu_s, g, n) of the volume

The resulting simulation files are: 
- MyVolumeName_output.jnii which contains the fluence data

Please let me know if I should provide an example.

**Comment by fangq** [comment]
@RecurvedBow, if you do not have any detector, or your detectors are not located within radius-distance away from the boundary, the saving detected photon flag will be turned off because there is nothing to be captured.

can you verify if you have detectors and they are located on the tissue boundary (volume bounding box, or non-zero-to-zero voxel boundary)?

**Reply by RecurvedBow** [reply]
Dear @fangq ,

Thank you for your help! I should have a detector placed on top of the volume by using "--saveref" and setting the mu_a and mu_s values to NaN at the top of the volume.  

I can confirm that I receive fluence data in MyVolumeName_output.jnii, from which I can extract reasonable reflectance values. I also get the same fluence data with and without using the args "--saveexit". My question is therefore where the photon data is supposed to be saved. There are not any further output files other than the files I have already described.

Please let me know whether I should provide further information.

**Reply by RecurvedBow** [reply]
Dear @fangq , 

I would like to ask you again whether you have any suggestions what I could check out.
